isogeny_primes.py

```diff
@@ -793,80 +842,273 @@ def get_aux_primes(K, norm_bound, C_K, h_K, contains_imaginary_quadratic):
     return aux_primes
 
 
-def get_AB_integers(embeddings,frak_q,epsilons,q_class_group_order):
+def filter_possible_values(possible_values_list, fq, prime_field):
+
+    output = []
+
+    for c in possible_values_list:
+        if c**2 == prime_field(1):
+            output.append(c)
+        elif c**2 == prime_field(fq.norm()):
+            output.append(c)
+        else:
+            fq_char = ZZ(fq.norm()).prime_divisors()[0]
+            possible_mid_coeffs = lifts_in_range(
+                2 * sqrt(fq.norm()), c + prime_field(fq.norm()) / c)
+            possible_weil_polys = [
+                x**2 + a * x + fq.norm() for a in possible_mid_coeffs
+            ]
+            elliptic_weil_polys = [
+                f for f in possible_weil_polys if weil_polynomial_is_elliptic(
```

Comment:
the lifts in range function due to the safety of +0.5 and -0.5 might actually cause some polynomials to not be a weil polynomial. Additionally the centered_lift might also fall outside the hasse interval if the characteristic of p >> 4*sqrt(fq.norm()) and the centered lift is always a possible mid value according to your algorithm.

There are two possible fixes for this:
1. make lifts_in_range such that you are always guaranteed to get a weil polynomial
  - make the computation exact by passing fq.norm() as first argument instead of 2 * sqrt(fq.norm())
  -  compare low_run**2 <= 4*fq.norm()
  -  return empty list if centred_lift > 4*fq.norm()
  
  
2. filter possible_weil_polyes using the builtin sage function .is_weil_polynomial()

Reply:
I think another way to fix `lifts_in_range`, which maintains its agnosticism about how it's actually being used (i.e. it stays as an elementary number theory algorithm), is to replace `return list(set(output))` with `return [a for a in list(set(output)) if a.abs() <= N]`. With this, `centered_lift` is no longer always in the output, and ensures the output is indeed in the range [-N,N], so the +/- 0.5 concern is also satisfied?

Reply:
yeah that would work, at least if sqrt(fq.norm()) is an exact element and not floating point. ps. why did you add the +/-0.5?

Reply:
I added +/- 0.5 to ensure that the limits -N and +N would be considered to be added to the output; I was concerned about using `>=` and `<=` and comparing floats, sage ints, and sage ints that are constructed as square roots of square numbers. However indeed, upon reconsideration and some testing of types, and as you are suggesting, this is unnecessary, and using  `>=` and `<=` seems fine.

Reply:
Fixed in latest commit